src/main/java/org/broadinstitute/consent/http/models/support/TicketFactory.java

```diff
@@ -28,37 +28,15 @@ public Request parseRequestResponse(String response) {
   /**
    * Constructs a DuosTicket with the proper structure to request support via Zendesk
    *
-   * @param name        The name of the user requesting support
-   * @param type        The type of request ("question", "incident", "problem", "task")
-   * @param email       The email of the user requesting support
-   * @param subject     Subject line of the request
-   * @param description Description of the task or question
-   * @param url         The origin url of this request
-   * @param uploads     Optional list of attachment tokens
+   * @param ticketFields  TicketFields
    */
-  public DuosTicket createTicket(String name, SupportRequestType type, String email, String subject,
-      String description, String url, List<String> uploads) {
-    if (name == null || email == null) {
-      throw new IllegalArgumentException("Name and email of user requesting support is required");
-    }
-    if (subject == null) {
-      throw new IllegalArgumentException("Support ticket subject is required");
-    }
-    if (description == null) {
-      throw new IllegalArgumentException("Support ticket description is required");
-    }
-    if (type == null) {
-      throw new IllegalArgumentException("Support ticket type is required");
-    }
-    if (url == null) {
-      throw new IllegalArgumentException("Support ticket url is required");
-    }
-
+  public DuosTicket createTicket(TicketFields ticketFields) {
+    ticketFields.validate();
     Ticket ticket = new Ticket(
-        new Ticket.Requester(name, email),
-        subject,
-        createComment(description, url, uploads));
-    ticket.setCustomFields(createCustomFields(name, type, email, description));
+        new Ticket.Requester(ticketFields.name(), ticketFields.email()),
+        ticketFields.subject(),
+        createComment(ticketFields.description(), ticketFields.url(), ticketFields.uploads()));
```

Comment:
When you're mapping from one object to another, I think it's best to put the mapping code inside either the target or source, if you have control over that code. In this case we have control over the source object, so this could be moved into `TicketFields`:
```
public record TicketFields(

...

  Ticket toTicket() {
    return new Ticket(new Ticket.Requester(name, email), subject, createComment());
  }

  private Comment createComment() {
    Comment comment = new Comment();
    comment.setBody(description + "\n\n------------------\nSubmitted from: " + url);
    if (uploads != null && !uploads.isEmpty()) {
      comment.setUploads(uploads);
    }
    return comment;
  }
```